app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,63 @@
+class ArticlesController < ApplicationController
+  before_action :set_article, only: [:edit, :update, :show, :destroy]
+  before_action :require_user
+  before_action :require_same_user, only: [:edit, :update, :destroy]
+  
+  def index
+    @articles = Article.paginate(page: params[:page], per_page: 5)
+  end
+  
+  def new
+    @article = Article.new
+  end
+  
+  def edit
+  end
+  
+  def create
+    @article = Article.new(article_params)
```

Comment:
Params sanitisation is not done. e.g some required params wouldn't have been passed and logic here seems to not handling that. Please check other cases here too along with the above quoted example.